### PR TITLE
[Hub Generated] Review request for Microsoft.Security to add version stable/2019-08-01

### PR DESCRIPTION
**Pull request by liranc**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/liranc/azure-rest-api-specs/alerts-2019-08-01...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotAlertTypes.json [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/iotSecuritySolutions/{solutionName}/analyticsModels/default/iotAlertTypes/{iotAlertTypeName}": {
```
Comment (majastrz) [inline specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotAlertTypes.json:37]: You should also have a path to list all alert types.

Comment (liranc) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotAlertTypes.json:37]: Is this a must-requirement?
If possible, we prefer to not expose a list path, but instead allow the user to query specific ones.

Comment (majastrz) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotAlertTypes.json:37]: Unfortunately, it is. Having a resource comes with certain expectations from the platform even if it's a read only resource. This is documented in the RPC repo at https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md. There are several services in Azure that rely on this to perform various automated scans. Weird deviations like this cause problems for them and also cause customer complaints about API inconsistencies.

Comment (liranc) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotAlertTypes.json]: Added List-AlertTypes path.

Comment (majastrz) [reply specification/security/resource-manager/Microsoft.Security/stable/2019-08-01/iotAlertTypes.json]: Thanks!

---
In reply to: [438729472](https://github.com/Azure/azure-rest-api-specs/pull/9653#discussion_r438729472) [](ancestors = 438729472)

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Comment by NullMDR** [comment]
@liranc Ready to merge?

**Comment by liranc** [comment]
> @liranc Ready to merge?

@nullmdr not quite yet (these changes aren't deployed yet).
Will ping you once we're ready.

Thanks!

**Comment by liranc** [comment]
@NullMDR we're ready now. Could you please merge the PR?

**Pull request merged**